Kudu.Core/Helpers/SiteTokenHelper.cs

```diff
@@ -0,0 +1,87 @@
+//------------------------------------------------------------------------------
+// <copyright file="SiteTokenHelper.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using Kudu.Core.Settings;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Kudu.Core.Helpers
+{
+    /// <summary>
+    /// Helper to issue and authenticate x-ms-site-token between antares role and site communication.
+    /// </summary>
+    public static class SiteTokenHelper
+    {
+        public const string SiteTokenHeader = "x-ms-site-token";
+
+        private static string _siteTokenAudience;
+        private static string _siteTokenIssuer;
+        private static TimeSpan? _siteTokenValidity;
+
+        // Setter for negative testing purposes
+        public static string SiteTokenAudience
+        {
+            get => _siteTokenAudience ?? "https://appservice.core.azurewebsites.net";
+            set => _siteTokenAudience = value;
+        }
+
+        public static string SiteTokenIssuer
+        {
+            get => _siteTokenIssuer ?? "https://appservice.core.azurewebsites.net";
```

Comment:
Should the issuer be https://<site>.scm.azurewebsites.net?

Reply:
Fixed.